ledger-tool/src/main.rs

```diff
@@ -728,13 +762,31 @@ fn open_blockstore(
     match Blockstore::open_with_options(
         ledger_path,
         BlockstoreOptions {
-            access_type,
-            recovery_mode: wal_recovery_mode,
+            access_type: access_type.clone(),
+            recovery_mode: wal_recovery_mode.clone(),
             enforce_ulimit_nofile: true,
             ..BlockstoreOptions::default()
         },
     ) {
         Ok(blockstore) => blockstore,
+        Err(BlockstoreError::RocksDb(err))
+            if err
+                .to_string()
+                .starts_with("Invalid argument: Column family not found:")
+                && access_type == AccessType::Secondary =>
+        {
+            info!("Blockstore is missing column(s) and unable to create with Secondary access");
+            info!("Attempting to temporarily open with Primary access to create missing column(s)");
+            open_blockstore_with_temporary_primary_access(
+                ledger_path,
+                access_type,
+                wal_recovery_mode,
+            )
+            .unwrap_or_else(|err| {
+                eprintln!("Failed to open ledger at {:?}: {:?}", ledger_path, err);
```

Comment:
can we make this error string unique?  it collides with the top-level open error, which might make debugging tricky

Reply:
> nice! wdyt about gating this behind something like `--force-open-blockstore`?

My initial thinking was to do this automatically given that `solana-validator` will update blockstore automatically. However, I'm amenable to the flag, and given that we're already detecting this exact scenario, we could print out a very helpful message if the flag isn't set.

> we could use the same logic for creating a new blockstore when called with an empty ledger dir?

Yessir, I can add logic to detect this error as well.

Reply:
I guess my thinking for gating it with a flag is from the scars of bricking more than one sole replicator ledgers with ledger-tool in the past.  seems like it'd be safer to explicitly request write ops